annotation-processor/src/main/java/org/hibernate/validator/ap/util/ConstraintHelper.java

```diff
@@ -361,6 +404,119 @@ public Boolean visitDeclared(DeclaredType constraintValidatorImplementation, Voi
 				)
 		);
 	}
+	
+	/**
+	 * Resolve the actual {@code AnnotationProcessorValidationTarget} of a constraint annotation, when applied to a method/constructor.
+	 * <p>
+	 * When the annotation supports multiple {@link AnnotationProcessorValidationTarget}s (i.e. it is both cross-parameter and generic), the actual target is resolved using the 
+	 * 'validationAppliesTo()' attribute of the annotation.
+	 * 
+	 * @param element  the method/constructor on which the annotation is applied
+	 * @param annotation  the constraint annotation
+	 * @return the resolved {@code AnnotationProcessorValidationTarget}, null if the target cannot be inferred
+	 */
+	public AnnotationProcessorValidationTarget resolveValidationTarget(ExecutableElement element, AnnotationMirror annotation) {
+		Set<AnnotationProcessorValidationTarget> allowedTargets = getSupportedValidationTargets( annotation.getAnnotationType() );
+		
+		// assume that at least one target (AnnotationProcessorValidationTarget.ANNOTATED_ELEMENT) is present
+		assert allowedTargets.size()>0;
```

Comment:
We usually don't use `assert` statements in the code base, as one cannot really rely on them (it's a user's choice whether to enable them or not). I think you can just remove it, given you obtain the value from another method of the same class so we really know it's never empty.